src/diffusers/pipelines/text_to_video_synthesis/pipeline_text_to_video_zero.py

```diff
@@ -0,0 +1,508 @@
+            prompt = [prompt]
+        if isinstance(negative_prompt, str):
+            negative_prompt = [negative_prompt]
+
+        # Default height and width to unet
+        height = height or self.unet.config.sample_size * self.vae_scale_factor
+        width = width or self.unet.config.sample_size * self.vae_scale_factor
+
+        # Check inputs. Raise error if not correct
+        self.check_inputs(prompt, height, width, callback_steps)
+
+        # Define call parameters
+        batch_size = 1 if isinstance(prompt, str) else len(prompt)
+        device = self._execution_device
+        # here `guidance_scale` is defined analog to the guidance weight `w` of equation (2)
+        # of the Imagen paper: https://arxiv.org/pdf/2205.11487.pdf . `guidance_scale = 1`
+        # corresponds to doing no classifier free guidance.
+        do_classifier_free_guidance = guidance_scale > 1.0
+
+        # Encode input prompt
+        text_embeddings = self._encode_prompt(
+            prompt, device, num_videos_per_prompt, do_classifier_free_guidance, negative_prompt
+        )
+
+        # Prepare timesteps
+        self.scheduler.set_timesteps(num_inference_steps, device=device)
+        timesteps = self.scheduler.timesteps
+
+        # Prepare latent variables
+        num_channels_latents = self.unet.in_channels
+
+        latents = self.prepare_latents(
+            batch_size * num_videos_per_prompt,
+            num_channels_latents,
+            1,
+            height,
+            width,
+            text_embeddings.dtype,
+            device,
+            generator,
+            latents,
+        )
+        dtype = latents.dtype
+
+        # when motion field is not used, augment with random latent codes
+        if use_motion_field:
+            latents = latents[:, :, :1]
+        else:
+            if latents.shape[2] < video_length:
+                latents_missing = self.prepare_latents(
+                    batch_size * num_videos_per_prompt,
+                    num_channels_latents,
+                    video_length - latents.shape[2],
+                    height,
+                    width,
+                    text_embeddings.dtype,
+                    device,
+                    generator,
+                    None,
+                )
+                latents = torch.cat([latents, latents_missing], dim=2)
+
+        t0 = timesteps[-t0].item()
+        t1 = timesteps[-t1].item()
+
+        x_t1_1 = None
+
+        # Prepare extra step kwargs.
+        extra_step_kwargs = self.prepare_extra_step_kwargs(generator, eta)
+        # Denoising loop
+        num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
+
+        shape = (batch_size, num_channels_latents, 1, height // self.vae_scale_factor, width // self.vae_scale_factor)
+
+        ddim_res = self.ddim_backward(
+            num_inference_steps=num_inference_steps,
+            timesteps=timesteps,
+            skip_t=1000,
```

Comment:
Should this configurable?

Reply:
Also, the block from L404 - L485 (which I believe is the meat of the pipeline) will benefit a lot from comments. Could we add them?

Reply:
Done!